src/OfflineTool.cpp

```diff
@@ -0,0 +1,503 @@
+        }
+        else
+            fail();
+    }
+    catch (std::exception const&)
+    {
+        fail();
+    }
+}
+
+void
+doCreateKeyfile(boost::filesystem::path const& keyFile,
+    boost::optional<std::string> const& keytype,
+    boost::optional<std::string> const& seed)
+{
+    using namespace ripple;
+    using namespace offline;
+
+    if (exists(keyFile))
+        throw std::runtime_error(
+            "Refusing to overwrite existing key file: " +
+            keyFile.string());
+
+    auto const kt = keytype ?
+        boost::optional<KeyType>{keyTypeFromString(*keytype)} :
+        boost::none;
+    if (kt && *kt == KeyType::invalid)
+    {
+        std::cerr << "Invalid key type: \"" << *keytype << "\"" <<
+            std::endl;
+        return;
+    }
+
+    auto key = RippleKey::make_RippleKey(kt, seed);
+
+    key.writeToFile(keyFile);
+
+    std::cout << "New ripple key created " <<
+        "in " << keyFile.string() << "\n" <<
+        "Key type is " << to_string(key.keyType()) << ", and " <<
+        "account ID is " <<
+        toBase58(calcAccountID(key.publicKey())) << "\n" <<
+        "\nThis file should be stored securely and not shared\n\n";
+
+}
+
+std::string
+getStdin()
+{
+    std::ostringstream stdinput;
+    stdinput << std::cin.rdbuf();
+    return stdinput.str();
+}
+
+void
+runCommand (const std::string& command,
+    std::vector <std::string> const& args,
+    boost::filesystem::path const& keyFile,
+    boost::optional<std::string> const& keyType,
+    InputType const& inputType)
+{
+    using namespace std;
+
+    struct commandParams
+    {
+        bool const allowNoInput;
+        std::function<void(boost::optional<std::string> const& input,
+            boost::filesystem::path const& keyFile,
+            boost::optional<std::string> const& keyType)> const action;
+    };
+    /* TODO: VC compiler doesn't like
+            std::function<void(std::string const& input)> const action;
+        with each of the lamdas capturing other local variables.
+    */
+    auto const serialize = [](
+        auto const& input, auto const&, auto const&)
+    {
+        BOOST_ASSERT(input);
+        doSerialize(*input);
+    };
+    auto const deserialize = [](
+        auto const& input, auto const&, auto const&)
+    {
+        BOOST_ASSERT(input);
+        doDeserialize(*input);
+    };
+    auto const sign = [](
+        auto const& input, auto const& keyFile, auto const&)
+    {
+        BOOST_ASSERT(input);
+        doSingleSign(*input, keyFile);
+    };
+    auto const multisign = [](
+        auto const& input, auto const& keyFile, auto const&)
+    {
+        BOOST_ASSERT(input);
+        doMultiSign(*input, keyFile);
+    };
+    auto const createkeyfile = [](
+        auto const& seed, auto const& keyFile, auto const& keyType)
+    {
+        doCreateKeyfile(keyFile, keyType, seed);
+    };
+    auto const argumenterror = []()
+        -> boost::none_t
```

Comment:
It doesn't compile under, I think, VS without a return type, which is why I added it.